src/messenger/messenger.cpp

```diff
@@ -148,7 +148,10 @@ class ZmqMessage {
       UNUSED_ATTRIBUTE int check =
           std::sscanf(payload_.c_str(), "%" SCNu64 "-%" SCNu64 "-", &send_msg_id_, &recv_cb_id_);
       NOISEPAGE_ASSERT(2 == check, "Couldn't parse the message header.");
-      message_.remove_prefix(message_.find_last_of('-') + 1);
+
```

Comment:
This allows for `-` in the message payload.

Reply:
yep, good catch